pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -295,27 +295,46 @@ def _check_config_and_set_final_flags(
             self._precision_flag = precision
 
         if plugins:
+            plugins_flags_count = {
+                "strategy": 0,
+                "precision_plugin": 0,
+                "precision": 0,
+                "checkpoint_io": 0,
+                "cluster_env": 0,
+            }
             for plugin in plugins:
                 if isinstance(plugin, Strategy) or isinstance(plugin, str) and plugin in self._registered_strategies:
                     self._strategy_flag = plugin
                     rank_zero_deprecation(
                         f"Passing {plugin} `strategy` to the `plugins` flag in Trainer has been deprecated"
                         f" in v1.5 and will be removed in v1.7. Use `Trainer(strategy={plugin})` instead."
                     )
+                    plugins_flags_count["strategy"] += 1
 
                 elif isinstance(plugin, PrecisionPlugin):
                     self._precision_plugin_flag = plugin
+                    plugins_flags_count["precision_plugin"] += 1
                 elif isinstance(plugin, str) and plugin in self._precision_types:
```

Comment:
when was passing `precision` to the `plugins` argument allowed? this is completely redundant with the `precision` flag on the Trainer constructor. I don't think this should be allowed

Reply:
But that would be a custom precision plug-in, right?

Was this supported?
```py
Trainer(plugins=["bf16"])
```

Reply:
No it is not. On 1.5 it errors. 
Sorry, I only saw `elif isinstance(plugin, PrecisionPlugin):` and your question, but you marked the line below which checks for string. My bad.

Reply:
In 1.5 we have this code:

```py
            if isinstance(plug, str) and plug in TrainingTypePluginsRegistry:
                if training_type is None:
                    training_type = TrainingTypePluginsRegistry.get(plug)
                else:
                    raise MisconfigurationException(
                        "You can only specify one precision and one training type plugin."
                        " Found more than 1 training type plugin:"
                        f' {TrainingTypePluginsRegistry[plug]["plugin"]} registered to {plug}'
                    )
            if isinstance(plug, str):
                # Reset the distributed type as the user has overridden training type
                # via the plugins argument
                self._distrib_type = None
                self.set_distributed_mode(plug)
```

which attempts to handle strings for training type plugins but there is no error handling for unsupported strings.

Reply:
Make sense. I will remove the precision str support and throw exceptions for non-supported strs

Reply:
Removed